docs/development/sync-protocol.md

```diff
@@ -0,0 +1,431 @@
+```
+
+#### `ruledownload` Response
+
+When a `ruledownload` request is received the sync server responds with a JSON object
+containing a list of rule objects and a cursor so the client can resume
+downloading if the rules need to be downloaded in multiple batches. 
+
+| Key | Required | Type | Meaning |
+|---|---|---|---|
+| cursor | NO | string | used to continue a rule download in a future request |
+| rules | YES | list of Rule objects | List of rule objects |
+
+##### Rules Objects
+
+
+:warning: Rule objects must have one of either the `identifier` or `sha256` fields.  The `sha256` field is supported for backwards compatability. 
+
+| Key | Required | Type | Meaning | Example Value | 
+|---|---|---|---|---|
+| identifier | YES | string | The attribute of the binary the rule should match on e.g. the team ID of a binary or sha256 hash value | "ff2a7daa4c25cbd5b057e4471c6a22aba7d154dadfb5cce139c37cf795f41c9c" |
+| sha256 | YES if `identifier` is not used | string | The sha256 of an attribute of the binary the rule should match on sha256 hash value of the binary or one of its signing certificates | "ff2a7daa4c25cbd5b057e4471c6a22aba7d154dadfb5cce139c37cf795f41c9c" 
+| policy | YES | string | identifies the action to perform in response to the rule matching must be one of the examples. | "ALLOWLIST","ALLOWLIST_COMPILER", "BLACKLIST", "BLOCKLIST", "REMOVE", "SILENT_BLACKLIST", "SILENT_BLOCKLIST", "WHITELIST", "WHITELIST_COMPILER"  |
+| rule_type | YES | string | identifies the type of rule must be one of he examples | "BINARY", "CERTIFICATE", "TEAMID" | 
+| custom_msg | NO | string | a custom message to display when the rule matches | "Hello" |
+| creation_time | NO | float64 | time the rule was created | 1573543803.349378 |
+| file_bundle_binary_count | NO | 
+
+
+##### Example `ruledownload` Response Payload
+
+```json
+{
+	"rules": [{
+		"identifier": "ff2a7daa4c25cbd5b057e4471c6a22aba7d154dadfb5cce139c37cf795f41c9c",
+		"rule_type": "CERTIFICATE",
+		"policy": "BLOCKLIST",
+		"custom_msg": "",
+		"creation_time": 1573543803.349378
+	}, {
+		"identifier": "233e741538e1cdf4835b3f2662e372cf0c2694b7e20b4e4663559c7fb0a9f234",
+		"rule_type": "BINARY",
+		"policy": "ALLOWLIST",
+		"custom_msg": "",
+		"creation_time": 1573572118.380034,
+		"file_bundle_binary_count": 13,
+		"file_bundle_hash": "7466e3687f540bcb7792c6d14d5a186667dbe18a85021857b42effe9f0370805"
+	}, 
+	{
+		"identifier": "EQHXZ8M8AV",
+		"rule_type": "TEAMID",
+		"policy": "ALLOWLIST",
+		"custom_msg": "Allow Software Google's Team ID",
+		"creation_time": 1576623399.151607,
+		"file_bundle_binary_count": 7,
+		"file_bundle_hash": "e4736dd3a731f5f71850984175c0ec54dcde06021af18f476eb480c707fbecda"
+	}],
+	"cursor": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXzfmdvb2dsZS5jb206YXBwbm90aHJyYAsSCUJsb2NrYWJsZSJANGYyYTA2MjY1ZjRiODQ2M2Y2YjI0MmNiZTMwMTNkMGZhNjlkNDUxNmI4OTU3Y2I3ZDAxZDcyMTJkM2NhZmZiNAwLEgRSdWxlGICA8Kehk9MKDBgAIAA="
+}
+```
+
+Old style rule
```

Comment:
What's this?

Reply:
It's a left over statement, that was supposed to show rules with the `sha256` payload. I'll delete it.